qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -79,6 +83,47 @@ def init():
 }
 
 
+def inject_userscripts():
+    """Register user javascript files with the global profiles."""
+    # The greasemonkey metadata block support in qtwebengine only starts at 5.8
+    # Otherwise have to handle injecting the scripts into the page at very
+    # early load, probs same place in view as the enableJS check.
+    if not qtutils.version_check('5.8'):
+        return
```

Comment:
How much work would it be to get it to run? Seeing that people on Debian/Ubuntu (and currently Gentoo) will most likely be using Qt 5.7.1 for some time, it'd be nice if it worked there too.

Reply:
Okay. The problem was that injecting the document-start scripts at the obvious place (ie loadStarted) caused the script to be injected in the soon-to-be-replaced javascript environment. So essentially the signal got fired before it made sense to act on it. Looking at my patch I was using on_url_changed() instead but I guess I needed to check that that didn't cause scripts to be injected multiple times in some situations.

I'm not sure how reasonable it is for a userscript to expect to be injected and run before the page loads at todays internet speeds though and I suspect that loading the script at any damn time we want wouldn't break much. If you depend on something you should set a timeout and poll for it like everyone else.

Reply:
Couldn't you still use [QWebEngineScript](https://doc.qt.io/qt-5/qwebenginescript.html) and set the [InjectionPoint](https://doc.qt.io/qt-5/qwebenginescript.html#InjectionPoint-enum) correctly, even when Qt doesn't handle the `@attributes` itself?